### PR TITLE
chore: register methods allowing to call enable/disable of the compat mode

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?

There are 2 commits but one is part of another PR: (not yet reviewed)
- [x] depends on https://github.com/containers/podman-desktop/pull/9611
need to rebase before merging

It allows to call with methods the enable or disable methods to enable or disable the compatibility mode on macOS

### What issues does this PR fix or reference?

related to #9242 

### How to test this PR?

unit test provided

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Comment by benoitf** [comment]
rebased

File: extensions/podman/packages/extension/src/extension.spec.ts [hunk 1]
```diff
@@ -2588,6 +2588,87 @@ async function testAudit(path: string, uri: string, condition: typeof expect | t
   expect(auditRecords.records).toEqual(condition.arrayContaining([expect.objectContaining({ type: 'error' })]));
 }
 
+test('activate on mac register commands for setting compatibility moide ', async () => {
+  vi.mocked(isMac).mockReturnValue(true);
+  vi.mocked(isWindows).mockReturnValue(false);
+  vi.mocked(isLinux).mockReturnValue(false);
+  vi.spyOn(PodmanInstall.prototype, 'checkForUpdate').mockResolvedValue({
+    hasUpdate: false,
+  } as unknown as UpdateCheck);
```
Comment (axel7083) [inline extensions/podman/packages/extension/src/extension.spec.ts:2597]: question: could this have side effect on the other tests ?

File: extensions/podman/packages/extension/src/extension.ts [hunk 2]
```diff
@@ -1390,8 +1391,23 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
       }
     });
 
+    // register two commands to enable and disable compatibility mode
+    extensionContext.subscriptions.push(
+      extensionApi.commands.registerCommand('podman.disableCompatibilityMode', async () => {
```
Comment (axel7083) [inline extensions/podman/packages/extension/src/extension.ts:1396]: question: should we had the command to the package.json ? other podman commands are listed in it.

**Review by axel7083 — APPROVED** [review]
LGTM code-wise. I do not see the commands added on windows (command palette) as the changes only target macOS. 

remark (not blocking): I am able to see the `podman.socketCompatibilityMode` command in the command palette even tho, it has not been registered and do nothing when I use it🤔 

image

File: extensions/podman/packages/extension/src/extension.spec.ts (hunk 1 above)
Comment (benoitf) [reply extensions/podman/packages/extension/src/extension.spec.ts:2597]: not more than the other https://github.com/containers/podman-desktop/blob/a230b9d8c2c3102df3df01955f4ac5134a860ca9/extensions/podman/packages/extension/src/extension.spec.ts#L2532

but anyway there is
https://github.com/containers/podman-desktop/blob/a230b9d8c2c3102df3df01955f4ac5134a860ca9/extensions/podman/packages/extension/src/extension.spec.ts#L176

so having all mocks being reset it should not impact any other test

File: extensions/podman/packages/extension/src/extension.ts (hunk 2 above)
Comment (benoitf) [reply extensions/podman/packages/extension/src/extension.ts:1396]: we don't need as we don't provide any shorcuts/nice display name etc, being mostly internal

**Comment by benoitf** [comment]
> remark (not blocking): I am able to see the podman.socketCompatibilityMode command in the command palette even tho, it has not been registered and do nothing when I use it🤔

it's registered there
https://github.com/containers/podman-desktop/blob/c8e8e4381afb489a6af2a1e5c09b05914dbd55df/extensions/podman/packages/extension/src/extension.ts#L1410

My PR is adding `podman.enableCompatibilityMode` and `podman.disableCompatibilityMode` (not `podman.socketCompatibilityMode`)


This part will be removed when all the settings will be done using the docker compliance page (but I can't already remove the code as the page is only at experimental stage)

**Review by cdrage — APPROVED** [review]
Pulled the PR and tested and was able to still enable / disable it fine. 

Going through the code it LGTM (more tests than actual code which is awesome!).

LGTM

**Pull request merged**